src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -32,10 +35,10 @@ protected function configure()
     {
         $this
             ->setName('security:encode-password')
-            ->setDescription('Encode a password.')
+            ->setDescription('Encodes a password.')
             ->addArgument('password', InputArgument::OPTIONAL, 'Enter a password')
-            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.')
-            ->addArgument('salt', InputArgument::OPTIONAL, 'Enter the salt you want to use to encode your password.')
+            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.', 'Symfony\Component\Security\Core\User\User')
+            ->addOption('empty-salt', null, InputOption::VALUE_NONE, 'Do not generate a salt or let the encoder generate one.')
             ->setHelp(<<<EOF
 
 The <info>%command.name%</info> command allows to encode a password using encoders
```

Comment:
I find the command help very confusing. Here it is a proposal for changing it:

```
The <info>%command.name%</info> command encodes passwords according to your
security configuration. This command is mainly used to generate passwords for
the <comment>in_memory</comment> user provider type and for changing passwords
in the database while developing the application.
 
Suppose that you have the following security configuration in your application:

<comment>
# app/config/security.yml
security:
    encoders:
        Symfony\Component\Security\Core\User\User: plaintext
        AppBundle\Entity\User: bcrypt
</comment>

If you execute the command non-interactively, the default Symfony User class
is used and a random salt is generated to encode the passwod:

  <info>php %command.full_name% --no-interactive [password]</info>

Pass the full user class path as the second argument to encode passwords for
your own entities:

  <info>php %command.full_name% --no-interactive [password] AppBundle\Entity\User</info>

Executing the command interactively allows you to provide a random salt for
encoding the password:

  <info>php %command.full_name% [password] AppBundle\Entity\User</info>

In case your encoder doesn't require a salt, add the <comment>empty-salt</comment> option:

  <info>php %command.full_name% --empty-salt [password] AppBundle\Entity\User</info> 

```